components/tools/OmeroPy/src/omero/util/temp_files.py

```diff
@@ -131,14 +131,33 @@ def tmpdir(self):
         """
         locktest = None
 
-        omerotemp = os.environ.get("OMERO_TEMPDIR", None)
-        homeprop = None
-        try:
-            homeprop = get_user_dir()
-        except:
-            pass  # ticket:3194, ticket:5583
-        tempprop = tempfile.gettempdir()
-        targets = [omerotemp, homeprop, tempprop]
+        # Read temporary  files directory from deprecated OMERO_TEMPDIR envvar
+        custom_tmpdir_deprecated = os.environ.get('OMERO_TEMPDIR', None)
+        if 'OMERO_TEMPDIR' in os.environ:
+            import warnings
+            warnings.warn(
+                "OMERO_TEMPDIR is deprecated. Use OMERO_TMPDIR instead.",
+                DeprecationWarning)
+
+        # Read temporary files directory from OMERO_TMPDIR envvar
+        default_tmpdir = None
+        if custom_tmpdir_deprecated:
+            default_tmpdir = path(custom_tmpdir_deprecated) / "omero" / "tmp"
+        custom_tmpdir = os.environ.get('OMERO_TMPDIR', default_tmpdir)
+
+        # List target base directories by order of precedence
+        targets = []
+        if custom_tmpdir:
+            targets.append(path(custom_tmpdir))
+        targets.append(get_omero_userdir() / "tmp")
+        targets.append(path(tempfile.gettempdir()) / "omero" / "tmp")
```

Comment:
RFE for discussion: Use `tempfile.mkdtmp()` to create a random temp directory instead of `tempfile.gettempdir()) / "omero" / "tmp"`?